### ISSUE TITLE
Upgrade to Spring Cloud 2020.0.0-M3

### ISSUE THREAD

**Issue by spencergibb** [spring-io/start.spring.io#493]
It is compatible with boot 2.4.0-M1

https://repo.spring.io/milestone/org/springframework/cloud/spring-cloud-dependencies/2020.0.0-M3/

**Comment by snicoll** [comment]
We've discussed this some more and while we agree that downgrading the Spring Boot milestone for users that select a Spring Cloud dependency, we believe it would be acceptable to do so if we changed the way milestones are presented on start.spring.io. Rather than displaying them as `2.4.0 (M1)` we could display them as `2.4.0 (milestone)`, potentially renaming `SNAPSHOT` to `snapshot` for consistency.

We also discussed the issue of users creating broken Spring Cloud-based projects with SB 2.4 snapshots. If we know stuff to be broken, we should do our best to reflect that so the plan is to disable Spring Cloud for non milestones of the 2.4 line. Of course, as soon as the situation is resolved (or has improved significantly enough for users to try) we can lift that restriction.
